### ISSUE TITLE
Best way to apply to all `rbi` files in the project

### ISSUE THREAD

**Issue by RobinDaugherty** [Shopify/rubocop-sorbet#94]
This is not strictly an issue with rubocop-sorbet, but I've been unable to find a general solution to this problem, and it keeps us from being able to use rubocop-sorbet.

The issue being that once `**/*.rbi` is added to the `Include` list for rubocop, it causes all of the other configured cops to run on rbi files, making a lot of complaints (and changes). And of course rbi generation going forward will continue to create them as-is, so in my mind it's not desirable to enforce the same rubocop rules within rbi files.

My preference would be to only include rbi files for the Sorbet cops, and exclude them for all other cops, but there's no elegant way to do this.

Can anyone provide some guidance on this? I can open a PR to add to the README if we can come up with a solution.

**Comment by Morriar** [comment]
👋 Hey Robin? 

We provide two config within this suite of cops:

1. [default.yml](https://github.com/Shopify/rubocop-sorbet/blob/main/config/default.yml): the config to use for your application files
2. [rbi.yml](https://github.com/Shopify/rubocop-sorbet/blob/main/config/rbi.yml): the config to use for your shim files (should only be enabled for the `sorbet/rbi/shims/` directory)

Which one is causing you trouble? From the [inclusion of `**/*.rbi`](https://github.com/Shopify/rubocop-sorbet/blob/main/config/rbi.yml#L7-L8) you mention it seems you're trying to enable the `rbi.yml` config?

**Comment by Morriar** [comment]
I think your top level Rubocop configuration should only use `default.yml` as it contains the settings expected for Ruby files.

If you want to apply the `rbi.yml` configuration you can add a `.rubocop.yml` in your `sorbet/rbi/` directory. You can find an [example in Tapioca](https://github.com/Shopify/tapioca/blob/main/sorbet/rbi/.rubocop.yml).

**Comment by RobinDaugherty** [comment]
That makes sense, though I think the `README` could be more clear about that.

But doesn't this leave us with `rbi`-related cops not seeing `rbi` files that are outside `sorbet/rbi`, including `Sorbet/ForbidRBIOutsideOfAllowedPaths`, whose _express purpose_ is to look for `rbi` files that exist outside of `sorbet/rbi`?

**Comment by Morriar** [comment]
The `default.yml` config already contains the cops that make sense for your top level. For example `Sorbet/ForbidRBIOutsideOfAllowedPaths` [is setup in `default.yml`](https://github.com/Shopify/rubocop-sorbet/blob/main/config/default.yml#L71-L78).

But yeah I agree we can make the README clearer. I'll push a change 👍

**Comment by Morriar** [comment]
I've been testing this with this setup:

```
.rubocop.yaml
Gemfile
lib
  file.rb
  file.rbi
sorbet
  rbi
    file.rbi
```

Where `.rubocop.yml` contains this configuration:

```yml
require:
  - rubocop-sorbet

inherit_gem:
  rubocop-sorbet: config/default.yml

AllCops:
  Include:
    - "**/*.rbi"
```

And `sorbet/rbi/.rubocop.yml` contains this:

```yml
require:
  - rubocop-sorbet

inherit_gem:
  rubocop-sorbet: config/rbi.yml
```

And finally running rubocop:

```
$ bundle exec rubocop

Inspecting 2 files
CC

Offenses:

lib/file.rbi:1:1: C: Sorbet/ForbidRBIOutsideOfAllowedPaths: RBI file path should match one of: sorbet/rbi/**
# typed: true
^
sorbet/rbi/file.rbi:3:1: C: [Correctable] Sorbet/SingleLineRbiClassModuleDefinitions: Empty class/module definitions in RBI files should be on a single line.
class Foo ...
^^^^^^^^^

2 files inspected, 2 offenses detected, 1 offense auto-correctable
```

Would that work with your project?

**Comment by Morriar** [comment]
👋 Hey Robin, 

I might be missing something from your configuration because I'm unable to reproduce what you're saying.

Let's take this [demo repo](https://github.com/Morriar/test-rubocop-sorbet) as example:

* We created a `.rubocop.yml` root configuration including both `.rb` and `.rbi` files
* As well as a `sorbet/rbi/.rubocop.yml` configuration file to import the rbi specific rules

As you can see when you run `bundle exec rubocop` you do get the expected errors:

```
$ bundle exec rubocop
Inspecting 3 files
CCC

Offenses:

lib/file.rb:1:1: C: [Correctable] Style/FrozenStringLiteralComment: Missing frozen string literal comment.
# typed: true
^

lib/file.rb:5:1: C: [Correctable] Layout/TrailingEmptyLines: 1 trailing blank lines detected.

lib/file.rbi:1:1: C: Sorbet/ForbidRBIOutsideOfAllowedPaths: RBI file path should match one of: sorbet/rbi/**
# typed: true
^

lib/file.rbi:4:7: C: [Correctable] Style/ClassAndModuleChildren: Use nested module/class definitions instead of compact style.
class Foo::Bar
      ^^^^^^^^

sorbet/rbi/file.rbi:3:1: C: [Correctable] Sorbet/SingleLineRbiClassModuleDefinitions: Empty class/module definitions in RBI files should be on a single line.
class Foo::Bar ...
^^^^^^^^^^^^^^

3 files inspected, 5 offenses detected, 4 offenses auto-correctable
```

As you can see, no `rbi.yml` related cops are reporting offenses from my normal Ruby files and `AllCops` offenses such as `Style/FrozenStringLiteralComment` or `Style/ClassAndModuleChildren` are not reported from inside the `sorbet/rbi` directory.

Note that a `Style/ClassAndModuleChildren` offense is still reported for `lib/file.rbi` because it is in the included path for `AllCops`, but this offense will disappear as soon as we move the `.rbi` file to the proper location as reported by `Sorbet/ForbidRBIOutsideOfAllowedPaths`.

Do you mind trying with this [demo repo](https://github.com/Morriar/test-rubocop-sorbet) and validate that we're seeing the same behavior?

I hope it helps 🙏

**Comment by RobinDaugherty** [comment]
Thank you @Morriar! I was able to verify that the demo repo works as expected. I want to deeper analyze the differences with my project, but I'll go ahead and close this for now. I really appreciate your time on this!

**Comment by RobinDaugherty** [comment]
I was able to give this another look with the benefit of more experience with Sorbet. The base issue that confused me is that I'm using `srb rbi` to generate my RBI files, and the rubocop configuration provided by this project for RBI files enforces a style that is very different from the RBI code generated by Sorbet. I understand now that Shopify is using Tapioca to generate RBI file, so that is not wholly unexpected. Rather than apply tens of thousands of autofixes and continue to maintain those (which I couldn't actually do because it crashed rubocop), I chose to add `sorbet/rbi` to the `Exclude` list.
